ambry-store/src/main/java/com/github/ambry/store/HybridCompactionPolicy.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.utils.Time;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * HybridCompactionPolicy will run compaction with StatsBasedCompactionPolicy more frequently and with CompactAllPolicy
+ * one time out of storeCompactionPolicySwitchPeriod.
+ */
+public class HybridCompactionPolicy implements CompactionPolicy{
```

Comment:
I don't feel good about the way we count the compaction round. All blobStore(partition) uses the same CompactionPolicy instance, they would interface each other and in some cases causes unexpected result. 

Assume we have 7 partitions in one host, and compaction manager trigger compaction on each one of them in the same order alway, p1, p2, p3...p7, p1, p2, p3... p7. In this case, P7 would always use CompactionAllPolicy and the other partitions would always use statsbased compaction policy.

I think what we need is a per partition HybridCompactionPolicy. It can't be shared by partitions, because it's no longer a stateless object.

Reply:
Thanks for the advise. Implemented with first approach that maintains a map which key is replicaId and value is the counter.